### PR TITLE
Adds a periodic check that triggers a disconnect if a half-open connection is detected.

### PR DESCRIPTION
**Pull request by gpolitis**

### CODE REVIEW

**Comment by gpolitis** [comment]
> Why is a separate timer necessary, doesn't PingManager fire `failed` callbacks if it times out?

@bgrozev that was my first thought, as well as Damyan's but strangely the answer is no. At least that is my conclusion after investigating a heapdump from such a zombie Jibri.

File: jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java [hunk 1]
```diff
@@ -853,4 +855,49 @@ public void pingFailed()
...
+            if (con != null && con.isConnected() && con.isAuthenticated())
+            {
+                long lastStanzaReceivedMs = con.getLastStanzaReceived();
+                if (lastStanzaReceivedMs > 0)
+                {
+                    long nowMs = System.currentTimeMillis();
+                    if (nowMs - lastStanzaReceivedMs > 2 * 1000 * DEFAULT_PING_INTERVAL_SECONDS)
+                    {
+                        logger.warn("Half-open XMPP connection detected, will trigger a disconnect.");
+                        con.disconnect();
```
Comment (damencho) [inline jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java:897]: When you trigger the disconnect, it reconnects again, right?

**Review by damencho — CHANGES_REQUESTED** [review]
Can you please fix the styling of the opening braces? 

And maybe introduce a constant for `2*1000*DEFAULT_PING_INTERVAL_SECONDS`

Thanks

File: jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java (hunk 1 above)
Comment (gpolitis) [reply jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java:901]: Correct 👍

**Review by damencho — APPROVED** [review]

**Pull request merged**